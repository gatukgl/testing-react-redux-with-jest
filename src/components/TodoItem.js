import React, { PropTypes } from 'react'
import style from './TodoItem.css'

const TodoItem = ({ text, complete }) => (
  <li className={complete ? style.complete : ''}>{text}</li>
)

TodoItem.PropTypes = {
  text: PropTypes.string.isRequired,
  complete: PropTypes.bool,
}

export default TodoItem
