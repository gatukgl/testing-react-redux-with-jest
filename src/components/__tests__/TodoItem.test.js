import React from 'react'
import { shallow } from 'enzyme' // similar to render()
import toJson from 'enzyme-to-json'
import TodoItem from '../TodoItem'

// shallow rendered testing
describe('TodoItem', () => {
  let component
  beforeEach(() => {
    // render TodoItem with text
    component = shallow(<TodoItem text="Drink coffee" />)
  })

  it('renders correct structure', () => {
    // component.is() come from "ENZYME"
    expect(component.is('li')).toBe(true) // have <li> ???
  })

  it('has complete class if receiving complete prop', () => {
    // to override component's props
    component.setProps({ complete: true })
    // component.hasClass() come from "ENZYME"
    expect(component.hasClass('complete')).toBe(true)
  })
})

// Snapshot testing
describe('Snapshot summary', () => {
  // if we change TodoItem component UI - we need to update by using "u" a snapshot also
  it('matches its snapshot - not complete', () => {
    const component = shallow(<TodoItem text="Not complete" />)

    // we dont need to include toJson(component)
    // because when we use .toMatchSnapshot()
    // will automatically call .toJson() following .jestrc configuration
    // "snapshotSerializers": ["enzyme-to-json/serializer"]
    expect(component).toMatchSnapshot()
  })
  it('matches its snapshot - omplete', () => {
    const component = shallow(<TodoItem text="complete" complete />) // render TodoItem with Hello text
    expect(component).toMatchSnapshot()
  })
})
