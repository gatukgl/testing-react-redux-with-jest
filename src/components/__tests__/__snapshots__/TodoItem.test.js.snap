// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot summary matches its snapshot - not complete 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <li
    className=""
>
    Not complete
</li>,
  "nodes": Array [
    <li
      className=""
>
      Not complete
</li>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TodoItem
        text="Not complete"
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "text": "Not complete",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <li
          className=""
>
          Not complete
</li>,
        "_debugID": 6,
        "_renderedOutput": <li
          className=""
>
          Not complete
</li>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TodoItem
    text="Not complete"
/>,
}
`;

exports[`Snapshot summary matches its snapshot - omplete 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <li
    className="complete"
>
    complete
</li>,
  "nodes": Array [
    <li
      className="complete"
>
      complete
</li>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TodoItem
        complete={true}
        text="complete"
/>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "complete": true,
          "text": "complete",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <li
          className="complete"
>
          complete
</li>,
        "_debugID": 8,
        "_renderedOutput": <li
          className="complete"
>
          complete
</li>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TodoItem
    complete={true}
    text="complete"
/>,
}
`;
